src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -203,26 +203,14 @@ BoundExpression MakeIsDeclarationPattern(SyntaxNode syntax, BoundExpression lowe
                 //     return tmp.HasValue;
                 // }
                 var tmpType = _factory.SpecialType(SpecialType.System_Nullable_T).Construct(type);
-                if (tmpType == loweredInput.Type)
-                {
-                    var value = _factory.Call(
-                        loweredInput,
-                        GetNullableMethod(syntax, tmpType, SpecialMember.System_Nullable_T_GetValueOrDefault));
-                    var asg2 = _factory.AssignmentExpression(loweredTarget, value);
-                    var result = requiresNullTest ? MakeNullableHasValue(syntax, loweredInput) : _factory.Literal(true);
-                    return _factory.MakeSequence(asg2, result);
-                }
-                else
-                {
-                    var tmp = _factory.SynthesizedLocal(tmpType, syntax);
-                    var asg1 = _factory.AssignmentExpression(_factory.Local(tmp), _factory.As(loweredInput, tmpType));
-                    var value = _factory.Call(
-                        _factory.Local(tmp),
-                        GetNullableMethod(syntax, tmpType, SpecialMember.System_Nullable_T_GetValueOrDefault));
-                    var asg2 = _factory.AssignmentExpression(loweredTarget, value);
-                    var result = MakeNullableHasValue(syntax, _factory.Local(tmp));
-                    return _factory.MakeSequence(tmp, asg1, asg2, result);
-                }
+                var tmp = _factory.SynthesizedLocal(tmpType, syntax);
+                var asg1 = _factory.AssignmentExpression(_factory.Local(tmp), tmpType == loweredInput.Type ? loweredInput : _factory.As(loweredInput, tmpType));
```

Comment:
>tmpType == loweredInput.Type [](start = 78, length = 28)

Should dynamic/tuple name/custom modifiers mismatch be significant for the purposes of this comparison?